src/Assets/Attributes.php

```diff
@@ -0,0 +1,199 @@
+            $this->asset->basename()
+        )->extractInfo();
+
+        $length = Arr::get($id3, 'playtime_seconds', 0);
+
+        return ['duration' => $length];
+    }
+
+    /**
+     * Get the attributes of an image.
+     *
+     * @return array
+     */
+    private function getImageAttributes()
+    {
+        // Since assets may be located on external platforms like Amazon S3, we can't simply
+        // grab the attributes. So we'll copy it locally and read the attributes from there.
+        $manager = new MountManager([
+            'source' => $this->asset->disk()->filesystem()->getDriver(),
+            'cache' => $cache = $this->getCacheFlysystem(),
+        ]);
+
+        $cachePath = "{$this->asset->containerId()}/{$this->asset->path()}";
+
+        if ($manager->has($destination = "cache://{$cachePath}")) {
+            $manager->delete($destination);
+        }
+
+        $manager->copy("source://{$this->asset->path()}", $destination);
+
+        try {
+            [$width, $height] = getimagesize($cache->getAdapter()->getPathPrefix().$cachePath);
+            $size = compact('width', 'height');
+        } catch (\Exception $e) {
+            $size = [];
+        } finally {
+            $cache->delete($cachePath);
+        }
+
+        return $size;
+    }
+
+    /**
+     * Get the attributes of an SVG.
+     *
+     * @return array
+     */
+    private function getSvgAttributes()
+    {
+        // Since assets may be located on external platforms like Amazon S3, we can't simply
+        // grab the attributes. So we'll copy it locally and read the attributes from there.
+        $manager = new MountManager([
+            'source' => $this->asset->disk()->filesystem()->getDriver(),
+            'cache' => $cache = $this->getCacheFlysystem(),
+        ]);
+
+        $cachePath = "{$this->asset->containerId()}/{$this->asset->path()}";
+
+        if ($manager->has($destination = "cache://{$cachePath}")) {
+            $manager->delete($destination);
+        }
+
+        $manager->copy("source://{$this->asset->path()}", $destination);
+
+        $svg = simplexml_load_file($cache->getAdapter()->getPathPrefix().$cachePath);
+
+        $cache->delete($cachePath);
+
+        if ($svg['width'] && $svg['height']
+            && is_numeric((string) $svg['width'])
+            && is_numeric((string) $svg['height'])) {
+            return ['width' => (float) $svg['width'], 'height' => (float) $svg['height']];
+        } elseif ($svg['viewBox']) {
+            [,,$width, $height] = preg_split('/[\s,]+/', $svg['viewBox'] ?: '');
+
+            return compact('width', 'height');
+        }
+
+        return ['width' => 300, 'height' => 150];
+    }
+
+    /**
+     * Get the attributes of a video.
+     *
+     * @return array
+     */
+    private function getVideoAttributes()
+    {
+        $id3 = GetId3::fromDiskAndPath(
+            $this->asset->container()->diskHandle(),
+            $this->asset->basename()
```

Comment:
This should be `path()`, which is the reason the tests were failing. Also if you uploaded into a subdirectory you'd see it fail in the CP too.